### PR TITLE
Expand health endpoint

### PR DESCRIPTION
**Pull request by kad-korpem**
# Description

Previously, the `/health` endpoint would simply respond with an HTTP 200 to indicate the server is up. However, when serving vector tiles, the server may be up while the separate tileserver is down or otherwise unreachable. In this case the `/health` endpoint should not return HTTP 200. The endpoint has been expanded such that when OGC API Tiles are configured, a call to `/health` relays a HEAD request to a specific (predetermined) tile, and returns the resulting status code. When OGC API Tiles are not configured, the previous behaviour remains.

## Type of change

(Remove irrelevant options)

- Improvement of existing feature
- Configuration change

# Checklist:

- [x] I've double-checked the code in this PR myself
- [x] I've left the code better than before ([boy scout rule](https://wiki.c2.com/?BoyScoutRule))
- [x] The code is readable, comments are added that explain hard or non-obvious parts.
- [x] I've expanded/improved the (unit) tests, when applicable
- [x] I've run (unit) tests that prove my solution works
- [x] There's no sensitive information like credentials in my PR

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -135,7 +140,22 @@ the defaults by providing your own spec using the `openapi-file` CLI flag.
 
 #### Health checks
 
-Health endpoint is available on `/health`.
+Health endpoint is available on `/health`. When the server is configured to serve OGC API Tiles, the health endpoint will check for the presence of a specific tile on the configured tileserver.
+When no OGC API Tiles are configured, the health endpoint will simply serve an HTTP 200 when called.
+
+By default, when OGC API Tiles are configured, the checked tile is determined from a fixed lookup table, based on the deepest zoomlevel of `EPSG:28992` configured (e.g., if the deepest zoomlevel
```
Comment (rkettelerij) [inline README.md:146]: > of `EPSG:28992` configured

configured kan denk ik weg (qua zinsopbouw)?

File: config/ogcapi_tiles.go [hunk 2]
```diff
@@ -127,3 +131,39 @@ type ZoomLevelRange struct {
...
+	13: {x: 4576, y: 2925},
+	14: {x: 9152, y: 5851},
+	15: {x: 18304, y: 11702},
+	16: {x: 36608, y: 23404},
+}
+
+// +kubebuilder:object:generate=true
+type HealthCheck struct {
+	// Projection (SRS/CRS) used for tile healthcheck
+	Srs string `yaml:"srs" json:"srs" default:"EPSG:28992" validate:"required,startswith=EPSG:"`
```
Comment (rkettelerij) [inline config/ogcapi_tiles.go:164]: Toevoegen 
```
+kubebuilder:validation:Pattern=`^EPSG:\d+$`
```

File: internal/engine/health.go [hunk 3]
```diff
@@ -2,10 +2,33 @@ package engine
...
+			target, _ = url.Parse(tilesConfig.DatasetTiles.TileServer.String() + *tilesConfig.DatasetTiles.HealthCheck.TilePath)
+		} else if tilesConfig.Collections != nil {
+			target, _ = url.Parse(tilesConfig.Collections[0].Tiles.GeoDataTiles.TileServer.String() + *tilesConfig.Collections[0].Tiles.GeoDataTiles.HealthCheck.TilePath)
+		}
+
+		e.Router.Get("/health", func(w http.ResponseWriter, _ *http.Request) {
+			resp, err := client.Head(target.String())
+			if err != nil {
+				// exact error is irrelevant for health monitoring
+				w.WriteHeader(http.StatusNotFound)
```
Comment (rkettelerij) [inline internal/engine/health.go:23]: Bij nader inzien misschien toch maar een log.println `healthcheck failed: tile <path> not found` toevoegen.

File: internal/ogc/tiles/main_test.go [hunk 4]
```diff
@@ -21,6 +21,8 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+var tilePath = "/foo/12/34/56"
```
Comment (rkettelerij) [inline internal/ogc/tiles/main_test.go:24]: dekken we met deze testen zowel de dataset als geodata/collection tiles healthcheck logica?

**Review by rkettelerij — APPROVED** [review]

File: README.md [hunk 5]
```diff
@@ -135,7 +140,22 @@ the defaults by providing your own spec using the `openapi-file` CLI flag.
...
+configured is 12, the path of the checked tile is `/NetherlandsRDNewQuad/12/1462/2288`). It is possible to override the tile to be checked, both in terms of projection (SRS/CRS) and specific tile path:
+
+```yaml
+ogcApi:
+  tiles:
+    healthCheck:
+      srs: EPSG:3035
```
Comment (roelarents) [inline README.md:153]: projection != tileMatrixSet

Comment (roelarents) [reply README.md:153]: nvm. config is ook op basis van srs ipv tms

File: config/ogcapi_tiles.go (hunk 2 above)
Comment (roelarents) [reply config/ogcapi_tiles.go:164]: we zitten hier niet op gitea... ;) 
```suggestion
	+kubebuilder:validation:Pattern=`^EPSG:\d+$`
	Srs string `yaml:"srs" json:"srs" default:"EPSG:28992" validate:"required,startswith=EPSG:"`
```

**Pull request merged**